Tests: [Errno 13] Permission denied: '/usr/share/jupyter/lab'

I'm trying to build jupyterlab for Fedora. I'm running the test in isolated chroot and I get this failure:

```
+ /usr/bin/python3 -m pytest -vv --maxfail=1
============================= test session starts ==============================
platform linux -- Python 3.6.4, pytest-3.2.3, py-1.5.2, pluggy-0.4.0 -- /usr/bin/python3
cachedir: .cache
rootdir: /builddir/build/BUILD/jupyterlab-0.31.8, inifile:
collecting ... collected 30 items

jupyterlab/tests/test_build_api.py::BuildAPITest::test_build FAILED

=================================== FAILURES ===================================
___________________________ BuildAPITest.test_build ____________________________

self = <jupyterlab.tests.test_build_api.BuildAPITest testMethod=test_build>

    def test_build(self):
>       resp = self.build_api.build()

jupyterlab/tests/test_build_api.py:53: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
jupyterlab/tests/test_build_api.py:19: in build
    return self._req('POST', '')
jupyterlab/tests/utils.py:126: in _req
    response.raise_for_status()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Response [500]>

    def raise_for_status(self):
        """Raises stored :class:`HTTPError`, if one occurred."""
    
        http_error_msg = ''
        if isinstance(self.reason, bytes):
            # We attempt to decode utf-8 first because some servers
            # choose to localize their reason strings. If the string
            # isn't utf-8, we fall back to iso-8859-1 for all other
            # encodings. (See PR #3538)
            try:
                reason = self.reason.decode('utf-8')
            except UnicodeDecodeError:
                reason = self.reason.decode('iso-8859-1')
        else:
            reason = self.reason
    
        if 400 <= self.status_code < 500:
            http_error_msg = u'%s Client Error: %s for url: %s' % (self.status_code, reason, self.url)
    
        elif 500 <= self.status_code < 600:
            http_error_msg = u'%s Server Error: %s for url: %s' % (self.status_code, reason, self.url)
    
        if http_error_msg:
>           raise HTTPError(http_error_msg, response=self)
E           requests.exceptions.HTTPError: 500 Server Error: [Errno 13] Permission denied: '/usr/share/jupyter/lab' for url: http://localhost:12341/a%40b/lab/api/build

/usr/lib/python3.6/site-packages/requests/models.py:935: HTTPError
---------------------------- Captured stderr setup -----------------------------
JupyterLab server extension not enabled, manually loading...
Serving notebooks from local directory: /tmp/tmpnfce09ae/notebooks
0 active kernels
The Jupyter Notebook is running at:
http://localhost:12341/a%40b/?token=...
Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
Forbidden
403 GET /a%40b/api/contents (::1) 2.06ms referer=None
----------------------------- Captured stderr call -----------------------------
Build failed, running a clean and rebuild
500 POST /a%40b/lab/api/build (::1): [Errno 13] Permission denied: '/usr/share/jupyter/lab'
[Errno 13] Permission denied: '/usr/share/jupyter/lab'
{
  "Host": "localhost:12341",
  "User-Agent": "python-requests/2.18.4",
  "Accept-Encoding": "gzip, deflate",
  "Accept": "*/*",
  "Connection": "keep-alive",
  "Authorization": "token 4d4419d6",
  "Content-Length": "0"
}
500 POST /a%40b/lab/api/build (::1) 2022.85ms referer=None
!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 1 failures !!!!!!!!!!!!!!!!!!!!
=========================== 1 failed in 3.23 seconds ===========================
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib64/python3.6/logging/__init__.py", line 994, in emit
    stream.write(msg)
  File "/usr/lib/python3.6/site-packages/_pytest/capture.py", line 251, in write
    self.buffer.write(obj)
ValueError: I/O operation on closed file
Call stack:
  File "/usr/lib64/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib64/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/builddir/build/BUILD/jupyterlab-0.31.8/jupyterlab/tests/utils.py", line 98, in start_thread
    app.start()
  File "/usr/lib/python3.6/site-packages/notebook/notebookapp.py", line 1641, in start
    self.cleanup_kernels()
  File "/usr/lib/python3.6/site-packages/notebook/notebookapp.py", line 1522, in cleanup_kernels
    self.log.info(kernel_msg % n_kernels)
Message: 'Shutting down 0 kernels'
Arguments: ()
```

I wonder why do tests want to write to that folder?

--- a/jupyterlab/tests/utils.py
+++ b/jupyterlab/tests/utils.py
@@ -40,6 +40,7 @@ class LabTestBase(NotebookTestBase):
         runtime_dir = cls.runtime_dir = tmp('runtime')
         lab_dir = cls.lab_dir = tmp('lab')
         lab_settings = cls.lab_settings = tmp('labsettings')
+        lab_workspaces = cls.lab_workspaces = tmp('labworkspaces')
         cls.notebook_dir = tmp('notebooks')
         cls.env_patch = patch.dict('os.environ', {
             'HOME': cls.home_dir,
@@ -71,7 +72,10 @@ class LabTestBase(NotebookTestBase):
 
         def start_thread():
             app = cls.notebook = LabApp(
+                app_dir=lab_dir,
                 port=cls.port,
+                user_settings_dir=lab_settings,
+                workspaces_dir=lab_workspaces,
                 port_retries=0,
                 open_browser=False,
                 config_dir=cls.config_dir,

